langchain/chat_models/base.py

```diff
@@ -92,20 +93,33 @@ def generate(
         )
         try:
             results = [
-                self._generate(m, stop=stop, run_manager=run_manager, **kwargs)
+                self._generate(
+                    m,
+                    stop=stop,
+                    run_manager=run_managers[0] if run_managers else None,
+                    **kwargs,
+                )
                 if new_arg_supported
                 else self._generate(m, stop=stop)
                 for m in messages
             ]
         except (KeyboardInterrupt, Exception) as e:
-            run_manager.on_llm_error(e)
+            for run_manager in run_managers:
```

Comment:
same comment as in js. We can do better error handling and send the error only into the one that errored

Reply:
Added for chat models